WcaOnRails/app/models/competition_event.rb

```diff
@@ -3,4 +3,19 @@ class CompetitionEvent < ApplicationRecord
   belongs_to :competition
   belongs_to :event
   has_many :registration_competition_events, dependent: :destroy
+  has_many :rounds, -> { order(:nth_round) }
+  accepts_nested_attributes_for :rounds, allow_destroy: true
+
+  validate do
+    remaining_rounds = rounds.reject(&:marked_for_destruction?)
+    if remaining_rounds.length > 0
+      expected_nth_rounds = (1..remaining_rounds.last.nth_round).to_a
+      actual_nth_rounds = remaining_rounds.map(&:nth_round)
+
+      missing_nth_rounds = expected_nth_rounds - actual_nth_rounds
+      if missing_nth_rounds.length > 0
```

Comment:
Gah! I knew I was missing the opposite check here. Your check looks like it would work, but would make it hard to write a good error message. I'm going to try something slightly different, more in the vein of what I was already doing here.